examples/README.md

```diff
@@ -260,3 +268,29 @@ Loaded key to 0x80000001
 ```
 
 The `keyload` tool takes only one argument, the filename of the stored key. Because the information what is key scheme (RSA or ECC) is contained within the key blob.
+
+## Storing keys into the TPM's NVRAM
+
+These examples demonstrates how to use the TPM as secure vault for keys. There are two programs, one to store a TPM key into the TPM's NVRAM and another to extract the key from the TPM's NVRAM. Both examples can use parameter encryption to protect from MITM attacks. The Non-volatile memory location is protected with a password authorization that is passed in encrypted form, when "-aes" or "-xor" is given on the commmand line.
+
+Before running the examples, make sure there is a keyblob.bin generated using the keygen tool. The key can be of any type, RSA, ECC or symmetric. The example will store the private and public part. In case of a symmetric key the public part is meta data from the TPM. How to generate a key you can see above, in the description of the keygen example.
+
+Typical output for storing and then reading an RSA key using parameter encryption:
+
+```
+$ ./examples/nvram/store -aes
+TPM2_StartAuthSession: sessionHandle 0x2000000
+Reading 840 bytes from keyblob.bin
+Storing key at TPM NV index 0x1800202 with password protection
+Public part = 616 bytes
+Private part = 222 bytes
+NV write succeeded
+
+$ ./examples/nvram/read 616 222 -aes
```

Comment:
Having to supply the size of each public/private part seems problematic. Why not just used `sizeof(TPM2B_PUBLIC)` and `sizeof(TPM2B_PRIVATE)` like the `readKeyBlob` does? If you want to followup with this enhancement in PR #143 that is fine.

Reply:
Good question. As far as I know, TPM2B_PUBLIC is of size capable to take in the largest possible public key. Therefore, I thought having an argument to specify the size is a good approach. Similar for TPM2B_PRIVATE. I will check this and if I can avoid these manual arguments I will as addition to 143 Thanks for the hint!

Reply:
@dgarske coming back to this and related to #143 

I noticed that writeKeyBlob and readKeyBlob used in nvram/read use sizeof(key->pub) instead of the actual size of the Key Public Part. While this is okay for storing a file to disk, I am not sure if this is okay for using the NVRAM that on some TPMs is few kB only.

Example below:

In both cases, writeKeyBlob will write 616 bytes of public area, while the actual public area for the symmetric key is only 50 bytes and for RSA is only 278 bytes.

Should I just use sizeof(TPM2B_PUBLIC) for nvram/read or I should try to change the write/readKeyBlob helpers?

```
dimitartomov@Dimitars-MacBook-Pro wolfTPM % ./examples/keygen/keygen -sym
TPM2.0 Key generation example
	Key Blob: keyblob.bin
	Algorithm: SYMCIPHER
		AES, 256bits, CTR mode
	Template: Default
	Use Parameter Encryption: NULL
Loading SRK: Storage 0x81000200 (282 bytes)
AES Symmetric template with 256 bits, Counter mode
Creating new SYMCIPHER key...
Created new key (pub 50, priv 158 bytes)
Wrote 776 bytes to keyblob.bin
dimitartomov@Dimitars-MacBook-Pro wolfTPM % ./examples/keygen/keygen -t  
TPM2.0 Key generation example
	Key Blob: keyblob.bin
	Algorithm: RSA
	Template: Default
	Use Parameter Encryption: NULL
Loading SRK: Storage 0x81000200 (282 bytes)
RSA template
Creating new RSA key...
Created new key (pub 278, priv 222 bytes)
Wrote 840 bytes to keyblob.bin
dimitartomov@Dimitars-MacBook-Pro wolfTPM %
```